build-tools/packages/bundle-size-tools/src/utilities/getChunkParsedSize.ts

```diff
@@ -14,17 +14,33 @@ export function getChunkParsedSize(stats: StatsCompilation, chunkId: string | nu
 		);
 	}
 
-	const matchingAsset = stats.assets.find((asset) => {
+	//Find all the assets that contain the chunk. Note: an asset may contain more than one chunk.
+	const matchingAssets = stats.assets.filter((asset) => {
 		// Make sure to only look at js files and not source maps (assumes source maps don't end in .js)
 		if (asset.name.endsWith(".js")) {
-			// Assumes only a single chunk per asset, this may not hold for all apps.
-			return asset.chunks?.[0] === chunkId;
+			// If the asset contains the chunk, it should be considered when calculating the total size.
+			return asset.chunks?.includes(chunkId);
 		}
 
 		return false;
 	});
 
-	// If there's no matching asset it could be that it was removed in the new version of the bundle, not necessarily an
-	// error. In that case return 0 as its size.
-	return matchingAsset?.size ?? 0;
```

Comment:
Reverting this change I made myself recently. After further inspection, the actual issue was somewhere else, and we should only be calling this function with parameters that come from the same report file, so a chunk (found in said file) should always be present in at least one asset.